test/test.js

```diff
@@ -292,16 +293,57 @@ describe('onclickoutside hoc', function() {
   });
 
   describe('with child rendering as null', function() {
-    var StatelessComponent = () => null;
```

Comment:
This should still work and be tested for: "not even a React component" stateless generators that are just functions with a single argument, returning JSX (or, createElement equiv of course) are still fully supported by React, so testing for that should *not* be removed.

Rather than rewriting this test, we should add the new code as a new test case in addition to the one that was already here. We might want to rename the tests a little so that testing the null-generator has a reasonable name, but almost any time you remove (even if by rewriting) a test case, you're removing the ability to catch regression